### ISSUE TITLE
feature-request: ng-entity-service enhancement

### ISSUE THREAD

**Issue by Badisi** [datorama/akita#352]
## I'm submitting a...
`[x] Feature request`

## Proposal
I love the idea behind **ng-entity-service** as it really abstract all the crud boilerplate we often have to write in RESTful services.

I also love the idea behind **ng-entity-service-loader** because I'm a frontend developer and UX is really driving my developments. Therefore, presenting a user with a loading indicator enhance his experience.

Thinking of it a bit further, I also find the Material Design specification very inspirational. Especially the [pattern](https://material.io/design/communication/confirmation-acknowledgement.html#acknowledgement) where users are presented with a retry snackbar in case of error.

It would be really nice to see **ng-entity-service-loader** evolve in something more enhanced. Something that would bring **loaders** and **notifiers** in the same place + some **retry**, **cancel** mechanism, to help building templates with better UX.

## In a nutshell
Being able to have a single accessor to use in template (like loader.loadersFor) to observe a service request lifecycle (loading, success, error) and being able to retry or cancel the request in case of error.

**Comment by NetanelBasal** [comment]
Sounds good. I don't have the time to work on it. Do you want to create a PR?
